File: backend/src/main/scala/com/habittracker/domain/Habit.scala
package com.habittracker.domain

import com.habittracker.api.dto.HabitType
import io.circe.Json

import java.time.{LocalDate, OffsetDateTime}
import java.util.UUID

case class Habit(
                  id: UUID,
                  userId: UUID,
                  title: String,
                  description: String,
                  color: Int,
                  habitType: Json,
                  createdAt: OffsetDateTime,
                  endDate: OffsetDateTime
                )

File: frontend/lib/models/habit.dart
import 'package:frontend/utils/habit_type.dart';

class Habit {
  final String title;
  final String? description;
  final int color;
  final HabitType habitType;
  final DateTime? endDate;
  final DateTime createdAt = DateTime.now();

  Habit(this.title, this.description, this.color, this.habitType, this.endDate);
}
